### PR TITLE
chore: add bzlmod support

### PR DESCRIPTION
**Pull request by cgrindel**
- Add bzlmod support to the main repository and to the example repositories.
- Refactor some of the utility functions in `integration_test_utils` into `no_deps_utils`. Allows the use of the utility functions in bzlmod extension implementation.

Related to #120.

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -36,7 +36,8 @@ updatesrc_update_all(
     name = "update_all",
     targets_to_run = [
         "@rules_bazel_integration_test//tools:update_deleted_packages",
-        "//doc:update",
+        # GH120: Enable once stardoc supports repo mapping.
+        # "//doc:update",
```
Comment (kormide) [inline BUILD.bazel:40]: You can do something like [this](https://github.com/aspect-build/bazel-lib/blob/main/lib/private/docs.bzl#L23) if you don't want to disable the target.

File: examples/custom_test_runner/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,56 @@
+module(
+    name = "custom_test_runner_example",
```
Comment (kormide) [inline examples/custom_test_runner/MODULE.bazel:2]: Note that unpublished modules don't need a `module` declaration, if you want to leave it out for simplicity.

**Review by kormide — APPROVED** [review]

File: examples/custom_test_runner/MODULE.bazel (hunk 2 above)
Comment (cgrindel) [reply examples/custom_test_runner/MODULE.bazel:2]: Thanks. I did it to give the workspace a name like I do in the `WORKSPACE` file.

File: BUILD.bazel (hunk 1 above)
Comment (cgrindel) [reply BUILD.bazel:40]: Great idea. I updated `cgrindel_bazel_starlib` to support this. Once the release gets into BCR, I will update this PR with the new version.

Comment (cgrindel) [reply BUILD.bazel]: Done.

**Pull request merged**